File: src/main/kotlin/dev/murad/shipping/capability/StallingCapability.java
package dev.murad.shipping.capability;


import net.minecraft.resources.ResourceLocation;
import net.neoforged.neoforge.capabilities.EntityCapability;
import org.jetbrains.annotations.Nullable;

public interface StallingCapability {

    EntityCapability<StallingCapability, @Nullable Void> STALLING_CAPABILITY = EntityCapability.createVoid(
            ResourceLocation.parse("item_handler"), //"create"
            StallingCapability.class
    );

    boolean isDocked();
    void dock(double x, double y, double z);
    void undock();

    boolean isStalled();
    void stall();
    void unstall();

    boolean isFrozen();
    void freeze();
    void unfreeze();
}

File: src/main/kotlin/dev/murad/shipping/capability/StallingCapability.kt
package dev.murad.shipping.capability

import net.minecraft.resources.ResourceLocation
import net.neoforged.neoforge.capabilities.EntityCapability


interface StallingCapability {

    fun isDocked(): Boolean
    fun dock(x: Double, y: Double, z: Double)
    fun undock()

    fun isStalled(): Boolean
    fun stall()
    fun unstall()

    fun isFrozen(): Boolean
    fun freeze()
    fun unfreeze()

    companion object {
        
        val STALLING_CAPABILITY: EntityCapability<StallingCapability, Void?> = EntityCapability.createVoid(
            ResourceLocation.parse("item_handler"),  //"create"
            StallingCapability::class.java
        )
    }
}
